elastic-apm-agent/src/main/java/co/elastic/apm/agent/premain/AgentMain.java

```diff
@@ -161,6 +165,24 @@ private synchronized static void loadAndInitializeAgent(String agentArguments, I
         }
     }
 
+
+    @SuppressWarnings("removal")
+    private static void securityManagerCheck() {
+        SecurityManager sm = System.getSecurityManager();
+        if (sm == null) {
+            return;
+        }
+        try {
+            sm.checkPermission(new AllPermission());
```

Comment:
What if someone has configured more fine-grained permissions, such as
```
  permission java.lang.RuntimePermission "setFactory";
  permission java.lang.RuntimePermission "getClassLoader";
  permission java.lang.RuntimePermission "setContextClassLoader";
  permission java.net.SocketPermission "*", "connect,resolve";
```

Reply:
I think here we can rely on the past experience of @eyalkoren with that: there are simply lots of them.
Even simple things like reading a system property need to be explicitly enabled when using fine-grained permissions, and some like `SocketPermission` might have their values depend on `server_url` in agent configuration that can't be read and thus be added to the policy snippet here.

If the need to provide fine-grained details arises, then we can do the extra effort of gathering (and maintaining) an exhaustive list.

Reply:
Trying to maintain a minimal set of permissions sounds like a terrible idea for me. Even if we compile such, any agent upgrade may break due to additional required permission. 
I am personally fine with issuing this warning if `AllPermission` is not granted. If the user does maintain a proper policy with minimal permissions, the agent will work as expected and this warning can be ignored. On the other hand, if we just check specific permission here (e.g. try to `getClassLoader` or read a system property) and the permission set in the policy is not exhaustive, there won't be a warning and the agent will not work as expected, which is worse.